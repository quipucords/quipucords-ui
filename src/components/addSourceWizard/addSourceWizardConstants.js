import React from 'react';
import AddSourceWizardStepOne from './addSourceWizardStepOne';
import AddSourceWizardStepTwo from './addSourceWizardStepTwo';
import AddSourceWizardStepThree from './addSourceWizardStepThree';
import { translate } from '../i18n/i18n';

/**
 * Add a source, wizard steps. Applies state for wizard behavior.
 *
 * @type {{component: React.ReactNode, enableNext: Function|*, name: Function|*, canJumpTo: Function|boolean,
 *    nextButtonText: Function|*, id: number|string, hideBackButton: Function|boolean,
 *    hideCancelButton: Function|boolean}[]}
 */
const addSourceWizardSteps = [
  {
    id: 1,
    component: <AddSourceWizardStepOne />,
    canJumpTo: ({ fulfilled, pending }) => fulfilled === false && pending === false,
    enableNext: ({ stepOneValid }) => stepOneValid === true,
    name: () => translate('form-dialog.title', { context: ['add-source', 'step'] })
  },
  {
    id: 2,
    component: <AddSourceWizardStepTwo />,
    canJumpTo: ({ fulfilled, pending, stepOneValid }) =>
      fulfilled === false && pending === false && stepOneValid === true,
    enableNext: ({ stepOneValid, stepTwoValid }) => (stepOneValid && stepTwoValid) || false,
    name: () => translate('form-dialog.title', { context: ['add-source', 'step', 'two'] }),
    nextButtonText: () => translate('form-dialog.label', { context: ['submit', 'add-source'] })
  },
  {
    id: 3,
    component: <AddSourceWizardStepThree />,
    canJumpTo: ({ stepOneValid, stepTwoValid }) => (stepOneValid && stepTwoValid) || false,
    enableNext: ({ fulfilled, pending }) => fulfilled === true || pending === true,
    hideBackButton: ({ fulfilled, pending }) => fulfilled === true || pending === true,
    hideCancelButton: ({ fulfilled, pending }) => fulfilled === true || pending === true,
    name: () => translate('form-dialog.title', { context: ['add-source', 'step', 'three'] }),
    nextButtonText: () => translate('form-dialog.label', { context: ['close'] })
  }
];

/**
 * Edit a source, wizard steps. Applies state for wizard behavior.
 *
 * @type {{component: React.ReactNode, enableNext: Function|*, name: Function|*, canJumpTo: Function|boolean,
 *    nextButtonText: Function|*, id: number|string, hideBackButton: Function|boolean,
 *    hideCancelButton: Function|boolean}[]}
 */
const editSourceWizardSteps = [
  {
    id: 1,
    component: <AddSourceWizardStepTwo />,
    canJumpTo: ({ fulfilled, pending }) => fulfilled === false && pending === false,
    enableNext: ({ stepTwoValid }) => stepTwoValid || false,
    name: () => translate('form-dialog.title', { context: ['add-source', 'step', 'two'] }),
    nextButtonText: () => translate('form-dialog.label', { context: ['submit', 'add-source'] })
  },
  {
    id: 2,
    component: <AddSourceWizardStepThree />,
    canJumpTo: ({ stepTwoValid }) => stepTwoValid || false,
    enableNext: ({ fulfilled, pending }) => fulfilled === true || pending === true,
    hideBackButton: ({ fulfilled, pending }) => fulfilled === true || pending === true,
    hideCancelButton: ({ fulfilled, pending }) => fulfilled === true || pending === true,
    name: () => translate('form-dialog.title', { context: ['add-source', 'step', 'three'] }),
    nextButtonText: () => translate('form-dialog.label', { context: ['close'] })
  }
];

export { addSourceWizardSteps, editSourceWizardSteps };
