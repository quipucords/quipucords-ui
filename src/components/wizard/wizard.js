import React from 'react';
import PropTypes from 'prop-types';
import { ModalVariant, Wizard as PfWizard } from '@patternfly/react-core';
import classNames from 'classnames';
import { Modal } from '../modal/modal';

/**
 * FixMe: PF Wizard no longer allows "appendTo"
 * activated here, https://github.com/patternfly/patternfly-react/pull/3102
 * and then removed with PF-React PR#4255
 */
/**
 * A PF Wizard wrapper
 *
 * @param {object} props
 * @param {string|object} props.className
 * @param {boolean} props.isForm
 * @param {boolean} props.isNavHidden
 * @param {boolean} props.isOpen
 * @param {Array} props.steps
 * @param {string} props.variant
 * @param {object} props.props
 * @returns {React.ReactNode}
 */
const Wizard = ({ className, isForm, isNavHidden, isOpen, steps, variant, ...props }) => {
  const cssClassName = classNames(
    'quipucords-wizard',
    { 'quipucords-wizard__hide-nav': isNavHidden === true },
    { 'quipucords-wizard__hide-nav-last': isForm === true },
    className
  );

  return (
    <Modal className="quipucords-modal__wizard" variant={variant} isOpen={isOpen} isContentOnly>
      <PfWizard className={cssClassName} steps={steps} {...props} />
    </Modal>
  );
};

/**
 * Prop types
 *
 * @type {{isOpen: boolean, variant: string, className: string|object, isNavHidden: boolean, isForm: boolean,
 *     steps: Array}}
 */
Wizard.propTypes = {
  className: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  isForm: PropTypes.bool,
  isNavHidden: PropTypes.bool,
  isOpen: PropTypes.bool,
  steps: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
      component: PropTypes.node.isRequired,
      canJumpTo: PropTypes.oneOfType([PropTypes.func, PropTypes.bool]),
      enableNext: PropTypes.oneOfType([PropTypes.func, PropTypes.bool]),
      hideBackButton: PropTypes.oneOfType([PropTypes.func, PropTypes.bool]),
      hideCancelButton: PropTypes.oneOfType([PropTypes.func, PropTypes.bool]),
      name: PropTypes.oneOfType([PropTypes.func, PropTypes.node]),
      nextButtonText: PropTypes.oneOfType([PropTypes.func, PropTypes.node])
    })
  ),
  variant: PropTypes.oneOf([...Object.values(ModalVariant)])
};

/**
 * Default props
 *
 * @type {{isOpen: boolean, variant: null, className: null, isNavHidden: boolean, isForm: boolean, steps: *[]}}
 */
Wizard.defaultProps = {
  className: null,
  isForm: false,
  isNavHidden: false,
  isOpen: false,
  steps: [],
  variant: null
};

export { Wizard as default, Wizard };
