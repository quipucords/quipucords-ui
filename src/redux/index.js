import React from 'react';
import { connect } from 'react-redux';
import { useLocation, useNavigate, useParams } from 'react-router-dom';
import { store } from './store';
import { reduxActions } from './actions';
import { reduxHelpers } from './common';
import { storeHooks } from './hooks';
import { reduxReducers } from './reducers';
import { reduxSelectors } from './selectors';
import { reduxTypes } from './constants';

/**
 * Patch withRouter for legacy components.
 *
 * @param {React.ReactNode} Component
 * @returns {React.ReactNode}
 */
const withRouter = Component => props => {
  const location = useLocation();
  const navigate = useNavigate();
  const params = useParams();
  return (
    <Component
      {...props}
      history={{ push: navigate }}
      location={location}
      navigate={navigate}
      params={params}
      router={{ location, navigate, params }}
    />
  );
};

const connectRouter = (mapStateToProps, mapDispatchToProps) => component =>
  withRouter(connect(mapStateToProps, mapDispatchToProps)(component));

export {
  connect,
  connectRouter,
  reduxActions,
  reduxHelpers,
  reduxReducers,
  reduxSelectors,
  reduxTypes,
  store,
  storeHooks
};
