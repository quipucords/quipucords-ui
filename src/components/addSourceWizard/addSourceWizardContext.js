import { useShallowCompareEffect } from 'react-use';
import { AlertVariant } from '@patternfly/react-core';
import { reduxTypes, storeHooks } from '../../redux';
import apiTypes from '../../constants/apiConstants';
import { translate } from '../i18n/i18n';

/**
 * Return an updated source. Display relative toast messaging after wizard closes.
 *
 * @param {object} options
 * @param {Function} options.t
 * @param {Function} options.useDispatch
 * @param {Function} options.useSelector
 * @returns {{}}
 */
const useGetAddSource = ({
  t = translate,
  useDispatch: useAliasDispatch = storeHooks.reactRedux.useDispatch,
  useSelector: useAliasSelector = storeHooks.reactRedux.useSelector
} = {}) => {
  const selectorResponse = useAliasSelector(({ addSourceWizard }) => addSourceWizard, {});
  const dispatch = useAliasDispatch();
  const { edit, error, errorMessage, fulfilled, show, source } = selectorResponse;

  useShallowCompareEffect(() => {
    if (show === false && (fulfilled || error)) {
      dispatch({
        type: reduxTypes.toastNotifications.TOAST_ADD,
        alertType: (error && AlertVariant.danger) || AlertVariant.success,
        header: t('toast-notifications.title_add-source_hidden', {
          context: [error && 'error', edit && 'edit']
        }),
        message: t('toast-notifications.description_add-source_hidden', {
          context: [error && 'error', edit && 'edit'],
          message: errorMessage,
          name: source?.[apiTypes.API_SUBMIT_SOURCE_NAME]
        })
      });
    }
  }, [dispatch, edit, error, fulfilled, show, source, t]);

  return selectorResponse;
};

const context = {
  useGetAddSource
};

export { context as default, context, useGetAddSource };
