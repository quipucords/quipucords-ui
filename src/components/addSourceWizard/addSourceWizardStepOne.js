import React from 'react';
import PropTypes from 'prop-types';
import { Form } from '@patternfly/react-core';
import { Radio } from '../form/radio';
import { connect, store, reduxSelectors, reduxTypes } from '../../redux';
import { FormGroup } from '../form/formGroup';
import { FormState } from '../formState/formState';
import apiTypes from '../../constants/apiConstants';
import { translate } from '../i18n/i18nHelpers';

class AddSourceWizardStepOne extends React.Component {
  isStepValid = ({ values }) => {
    store.dispatch({
      type: reduxTypes.sources.VALID_SOURCE_WIZARD_STEPONE,
      source: {
        [apiTypes.API_SUBMIT_SOURCE_SOURCE_TYPE]: values.sourceType
      }
    });
  };

  render() {
    const { type, t } = this.props;

    return (
      <FormState validateOnMount setValues={{ sourceType: type }} validate={this.isStepValid}>
        {({ values, handleOnEvent, handleOnSubmit }) => (
          <Form isHorizontal onSubmit={handleOnSubmit}>
            <FormGroup role="radiogroup" isStack label={t('form-dialog.label', { context: 'source-type' })}>
              <Radio
                name="sourceType"
                id="sourceType-network"
                value="network"
                checked={values.sourceType === 'network'}
                onChange={handleOnEvent}
                label={t('form-dialog.label', { context: 'network-range' })}
              />
              <Radio
                name="sourceType"
                id="sourceType-openshift"
                value="openshift"
                checked={values.sourceType === 'openshift'}
                onChange={handleOnEvent}
                label={t('form-dialog.label', { context: 'openshift' })}
              />
              <Radio
                name="sourceType"
                id="sourceType-satellite"
                value="satellite"
                checked={values.sourceType === 'satellite'}
                onChange={handleOnEvent}
                label={t('form-dialog.label', { context: 'satellite' })}
              />
              <Radio
                name="sourceType"
                id="sourceType-vcenter"
                value="vcenter"
                checked={values.sourceType === 'vcenter'}
                onChange={handleOnEvent}
                label={t('form-dialog.label', { context: 'vcenter-server' })}
              />
              <Radio
                name="sourceType"
                id="sourceType-ansible"
                value="ansible"
                checked={values.sourceType === 'ansible'}
                onChange={handleOnEvent}
                label={t('form-dialog.label', { context: 'ansible-server' })}
              />
            </FormGroup>
          </Form>
        )}
      </FormState>
    );
  }
}

AddSourceWizardStepOne.propTypes = {
  type: PropTypes.string,
  t: PropTypes.func
};

AddSourceWizardStepOne.defaultProps = {
  type: 'network',
  t: translate
};

const makeMapStateToProps = () => {
  const mapSource = reduxSelectors.sources.makeSourceDetail();

  return (state, props) => ({
    ...mapSource(state, props)
  });
};

const ConnectedAddSourceWizardStepOne = connect(makeMapStateToProps)(AddSourceWizardStepOne);

export { ConnectedAddSourceWizardStepOne as default, ConnectedAddSourceWizardStepOne, AddSourceWizardStepOne };
