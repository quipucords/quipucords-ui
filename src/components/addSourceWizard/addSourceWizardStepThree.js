import React from 'react';
import PropTypes from 'prop-types';
import { EmptyState, EmptyStateBody, EmptyStateIcon, Spinner, Title } from '@patternfly/react-core';
import { OutlinedCheckCircleIcon, ExclamationCircleIcon } from '@patternfly/react-icons';
import { global_success_color_100 as green, global_danger_color_100 as red } from '@patternfly/react-tokens';
import { connect, reduxSelectors } from '../../redux';
import { EMPTY_CONTEXT, translate } from '../i18n/i18n';

/**
 * Add source wizard, step three, confirmation.
 *
 * @param {object} props
 * @param {boolean} props.add
 * @param {boolean} props.error
 * @param {boolean} props.fulfilled
 * @param {boolean} props.pending
 * @param {string} props.name
 * @param {Function} props.t
 * @returns {React.ReactNode}
 */
const AddSourceWizardStepThree = ({ add, error, fulfilled, pending, name, t }) => (
  <React.Fragment>
    <EmptyState className="quipucords-empty-state">
      {error && <EmptyStateIcon icon={ExclamationCircleIcon} color={red.value} />}
      {fulfilled && <EmptyStateIcon icon={OutlinedCheckCircleIcon} color={green.value} />}
      {pending && <EmptyStateIcon icon={Spinner} />}
      <Title headingLevel="h3">
        {t('form-dialog.empty-state_title_add-source', {
          context: [(error && 'error') || (pending && 'pending'), !add && 'edit'],
          name
        })}
      </Title>
      <EmptyStateBody>
        {t(`form-dialog.empty-state_description_add-source`, {
          context: [(error && 'error') || (pending && 'pending'), (!add && pending && 'edit') || EMPTY_CONTEXT],
          name
        })}
      </EmptyStateBody>
    </EmptyState>
  </React.Fragment>
);

/**
 * Prop types
 *
 * @type {{add: boolean, t: Function, pending: boolean, fulfilled: boolean, name: string, error: boolean}}
 */
AddSourceWizardStepThree.propTypes = {
  add: PropTypes.bool,
  error: PropTypes.bool,
  fulfilled: PropTypes.bool,
  pending: PropTypes.bool,
  name: PropTypes.string,
  t: PropTypes.func
};

/**
 * Default props
 *
 * @type {{add: boolean, t: translate, pending: boolean, fulfilled: boolean, name: null, error: boolean}}
 */
AddSourceWizardStepThree.defaultProps = {
  add: false,
  error: false,
  fulfilled: false,
  pending: false,
  name: null,
  t: translate
};

const makeMapStateToProps = () => {
  const mapSource = reduxSelectors.sources.makeSourceDetail();

  return (state, props) => ({
    ...state.addSourceWizard,
    ...mapSource(state, props)
  });
};

const ConnectedAddSourceWizardStepThree = connect(makeMapStateToProps)(AddSourceWizardStepThree);

export { ConnectedAddSourceWizardStepThree as default, ConnectedAddSourceWizardStepThree, AddSourceWizardStepThree };
