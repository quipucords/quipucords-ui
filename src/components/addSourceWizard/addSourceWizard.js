import React from 'react';
import PropTypes from 'prop-types';
import { ModalVariant } from '@patternfly/react-core';
import { reduxActions, reduxTypes, storeHooks } from '../../redux';
import { addSourceWizardSteps, editSourceWizardSteps } from './addSourceWizardConstants';
import { Wizard } from '../wizard/wizard';
import { useGetAddSource } from './addSourceWizardContext';
import apiTypes from '../../constants/apiConstants';
import { EMPTY_CONTEXT, translate } from '../i18n/i18n';

/**
 * Add a source with a Wizard.
 *
 * @fires onCancel
 * @fires onSubmit
 * @param {object} props
 * @param {Function} props.addSource
 * @param {Array} props.addSteps
 * @param {Array} props.editSteps
 * @param {Function} props.t
 * @param {Function} props.updateSource
 * @param {Function} props.useDispatch
 * @param {Function} props.useGetAddSource
 * @returns {React.ReactNode}
 */
const AddSourceWizard = ({
  addSource,
  addSteps,
  editSteps,
  t,
  updateSource,
  useDispatch: useAliasDispatch,
  useGetAddSource: useAliasGetAddSource
}) => {
  const dispatch = useAliasDispatch();
  const { edit, errorStatus, fulfilled, pending, source, stepOneValid, stepTwoValid, show } = useAliasGetAddSource();

  /**
   * If value is a function, run it and pass state for step validation.
   *
   * @param {*} value
   * @returns {*}
   */
  const isFunctionRun = value => {
    if (typeof value === 'function') {
      return value({ fulfilled, pending, stepOneValid, stepTwoValid });
    }
    return value;
  };

  const updatedSteps = ((edit && editSteps) || addSteps).map(step => {
    const updatedStep = {};
    Object.entries(step).forEach(([key, value]) => {
      updatedStep[key] = isFunctionRun(value);
    });
    return updatedStep;
  });

  /**
   * On close, or cancel the wizard process dispatch notifications, close the wizard.
   *
   * @event onCancel
   */
  const onCancel = () => {
    const closeWizard = () => {
      dispatch([
        {
          type: reduxTypes.confirmationModal.CONFIRMATION_MODAL_HIDE
        },
        {
          type: reduxTypes.sources.UPDATE_SOURCE_HIDE
        }
      ]);

      if (fulfilled) {
        dispatch({
          type: reduxTypes.sources.UPDATE_SOURCES
        });
      }
    };

    if (fulfilled || errorStatus >= 500 || errorStatus === 0) {
      closeWizard();
      return;
    }

    dispatch({
      type: reduxTypes.confirmationModal.CONFIRMATION_MODAL_SHOW,
      title: t('form-dialog.confirmation_title_add-source', { context: [pending && 'exit', edit && 'edit'] }),
      heading: t('form-dialog.confirmation_heading_add-source', { context: [pending && 'exit', edit && 'edit'] }),
      body: t('form-dialog.confirmation_body_add-source', {
        context: [pending && 'exit', edit && 'edit', !pending && EMPTY_CONTEXT]
      }),
      cancelButtonText: t('form-dialog.label', { context: 'no' }),
      confirmButtonText: t('form-dialog.label', { context: 'yes' }),
      onConfirm: closeWizard,
      variant: ModalVariant.medium
    });
  };

  /**
   * On save, submit, the wizard form.
   *
   * @event onSubmit
   */
  const onSubmit = () => {
    if (stepOneValid && stepTwoValid) {
      if (edit) {
        updateSource(source[apiTypes.API_SUBMIT_SOURCE_ID], source)(dispatch);
      } else {
        addSource(source, { scan: true })(dispatch);
      }
    }
  };

  return (
    <Wizard
      className="quipucords-wizard__add-source"
      isForm
      onClose={onCancel}
      onNext={onSubmit}
      isOpen={show}
      steps={updatedSteps}
      title={t('form-dialog.title', { context: ['add-source', edit && 'edit'] })}
      variant={ModalVariant.medium}
    />
  );
};

/**
 * Prop types
 *
 * @type {{editSteps: Array, edit: boolean, pending: boolean, stepOneValid: boolean, show: boolean,
 *     errorMessage: string, fulfilled: boolean, stepTwoValid: boolean, addSteps: Array, source: object,
 *     addSource: Function, updateSource: Function, t: Function, useDispatch: Function,
 *     errorStatus: number}}
 */
AddSourceWizard.propTypes = {
  addSource: PropTypes.func,
  addSteps: PropTypes.array,
  editSteps: PropTypes.array,
  t: PropTypes.func,
  updateSource: PropTypes.func,
  useDispatch: PropTypes.func,
  useGetAddSource: PropTypes.func
};

/**
 * Default props
 *
 * @type {{editSteps: *[], edit: boolean, pending: boolean, stepOneValid: boolean, errorMessage: null,
 *     fulfilled: boolean, stepTwoValid: boolean, addSteps: *[], source: {}, addSource: Function,
 *     updateSource: Function, t: translate, useDispatch: Function, errorStatus: null}}
 */
AddSourceWizard.defaultProps = {
  addSource: reduxActions.sources.addSource,
  addSteps: addSourceWizardSteps,
  editSteps: editSourceWizardSteps,
  t: translate,
  updateSource: reduxActions.sources.updateSource,
  useDispatch: storeHooks.reactRedux.useDispatch,
  useGetAddSource
};

export { AddSourceWizard as default, AddSourceWizard };
