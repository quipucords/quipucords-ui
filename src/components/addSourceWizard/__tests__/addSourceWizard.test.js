import React from 'react';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { store } from '../../../redux';
import { AddSourceWizard } from '../addSourceWizard';
import { Wizard } from '../../wizard/wizard';

describe('AddSourceWizard Component', () => {
  let mockDispatch;

  beforeEach(() => {
    mockDispatch = jest.spyOn(store, 'dispatch').mockImplementation((type, data) => ({ type, data }));
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  const generateEmptyStore = (obj = {}) => configureMockStore()(obj);

  it('should render a basic component', async () => {
    const props = {
      useGetAddSource: () => ({
        show: true
      })
    };

    const component = await shallowHookComponent(<AddSourceWizard {...props} />);
    expect(component).toMatchSnapshot('basic');
  });

  it('should display update steps', async () => {
    const props = {
      useGetAddSource: () => ({
        show: true,
        edit: true
      })
    };

    const component = await shallowHookComponent(<AddSourceWizard {...props} />);
    expect(component).toMatchSnapshot('update');
  });

  it('should not display a wizard', async () => {
    const props = {
      useGetAddSource: () => ({
        show: false
      })
    };

    const component = await mountHookComponent(<AddSourceWizard {...props} />);
    expect(component.find(Wizard).props()?.isOpen).toBe(false);
  });

  it('should allow cancelling the wizard', async () => {
    const mockStore = generateEmptyStore({
      addSourceWizard: {}
    });
    const props = {
      useGetAddSource: () => ({
        show: true
      })
    };

    const component = await mountHookComponent(
      <Provider store={mockStore}>
        <AddSourceWizard {...props} />
      </Provider>
    );

    component.find('button.pf-c-button.pf-m-link').first().simulate('click');
    expect(mockDispatch.mock.calls).toMatchSnapshot('cancel');
  });

  it('should allow closing the wizard on fulfillment', async () => {
    const mockStore = generateEmptyStore({
      addSourceWizard: {}
    });
    const props = {
      useGetAddSource: () => ({
        show: true,
        fulfilled: true
      })
    };

    const component = await mountHookComponent(
      <Provider store={mockStore}>
        <AddSourceWizard {...props} />
      </Provider>
    );

    component.find('button.pf-c-button.pf-m-link').first().simulate('click');
    expect(mockDispatch.mock.calls).toMatchSnapshot('fulfill');
  });
});
