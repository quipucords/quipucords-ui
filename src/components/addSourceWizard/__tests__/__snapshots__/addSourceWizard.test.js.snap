// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSourceWizard Component should allow cancelling the wizard: cancel 1`] = `
[
  [
    {
      "source": {
        "source_type": "network",
      },
      "type": "VALID_SOURCE_WIZARD_STEPONE",
    },
  ],
  [
    {
      "body": "t(form-dialog.confirmation_body_add-source, {"context":" "})",
      "cancelButtonText": "t(form-dialog.label, {"context":"no"})",
      "confirmButtonText": "t(form-dialog.label, {"context":"yes"})",
      "heading": "t(form-dialog.confirmation_heading_add-source, {"context":""})",
      "onConfirm": [Function],
      "title": "t(form-dialog.confirmation_title_add-source, {"context":""})",
      "type": "CONFIRMATION_MODAL_SHOW",
      "variant": "medium",
    },
  ],
]
`;

exports[`AddSourceWizard Component should allow closing the wizard on fulfillment: fulfill 1`] = `
[
  [
    {
      "source": {
        "source_type": "network",
      },
      "type": "VALID_SOURCE_WIZARD_STEPONE",
    },
  ],
  [
    [
      {
        "type": "CONFIRMATION_MODAL_HIDE",
      },
      {
        "type": "UPDATE_SOURCE_HIDE",
      },
    ],
  ],
  [
    {
      "type": "UPDATE_VIEW",
      "viewId": "sources",
    },
  ],
]
`;

exports[`AddSourceWizard Component should display update steps: update 1`] = `
<Wizard
  className="quipucords-wizard__add-source"
  isForm={true}
  isNavHidden={false}
  isOpen={true}
  onClose={[Function]}
  onNext={[Function]}
  steps={
    [
      {
        "canJumpTo": false,
        "component": <Memo(Connect(AddSourceWizardStepTwo)) />,
        "enableNext": false,
        "id": 1,
        "name": "t(form-dialog.title_add-source_step, {"context":"two"})",
        "nextButtonText": "t(form-dialog.label_submit, {"context":"add-source"})",
      },
      {
        "canJumpTo": false,
        "component": <Memo(Connect(AddSourceWizardStepThree)) />,
        "enableNext": false,
        "hideBackButton": false,
        "hideCancelButton": false,
        "id": 2,
        "name": "t(form-dialog.title_add-source_step, {"context":"three"})",
        "nextButtonText": "t(form-dialog.label, {"context":"close"})",
      },
    ]
  }
  title="t(form-dialog.title_add-source, {"context":"edit"})"
  variant="medium"
/>
`;

exports[`AddSourceWizard Component should render a basic component: basic 1`] = `
<Wizard
  className="quipucords-wizard__add-source"
  isForm={true}
  isNavHidden={false}
  isOpen={true}
  onClose={[Function]}
  onNext={[Function]}
  steps={
    [
      {
        "canJumpTo": false,
        "component": <Memo(Connect(AddSourceWizardStepOne)) />,
        "enableNext": false,
        "id": 1,
        "name": "t(form-dialog.title_add-source, {"context":"step"})",
      },
      {
        "canJumpTo": false,
        "component": <Memo(Connect(AddSourceWizardStepTwo)) />,
        "enableNext": false,
        "id": 2,
        "name": "t(form-dialog.title_add-source_step, {"context":"two"})",
        "nextButtonText": "t(form-dialog.label_submit, {"context":"add-source"})",
      },
      {
        "canJumpTo": false,
        "component": <Memo(Connect(AddSourceWizardStepThree)) />,
        "enableNext": false,
        "hideBackButton": false,
        "hideCancelButton": false,
        "id": 3,
        "name": "t(form-dialog.title_add-source_step, {"context":"three"})",
        "nextButtonText": "t(form-dialog.label, {"context":"close"})",
      },
    ]
  }
  title="t(form-dialog.title, {"context":"add-source"})"
  variant="medium"
/>
`;
